src/compiler/conditionLifting.ml

```diff
@@ -289,6 +261,52 @@ let prov_transitivity (provs : prov_exprs) : prov_exprs =
   Variable.Map.fold (fun dest _se tprovs -> fst @@ memoize tprovs dest)
     provs Variable.Map.empty
 
+(* The following transforms event expression into equations using previously
+   definied provenance.
+
+   An important note: there is an hidden hypothesis on the way equations are
+   specified by the user. [v = v'] as an event really means "when v reach v'",
+   implying that [v] is assumed to be initially lower than [v']. This is
+   important because of the semantics of threshold passing: the event is deemed
+   reached when [v >= v'], and not reach yet (or reverted!) when [v < v']. *)
```

Comment:
The whole point of this pass is to eliminate hysteresis entirely.

I'd argue that allowing negative values is not a good idea, I feel that what we currently have works as well as it does because we keep everything in the positive. But this can be something we can spend time thinking about if it appears to make sense from a UX perspective.